OpenRA.Mods.Common/Traits/Buildings/Building.cs

```diff
@@ -40,6 +36,9 @@ public class BuildingInfo : ITraitInfo, IOccupySpaceInfo, IPlaceBuildingDecorati
 			"Set to '-1' to disable adjacency checks.")]
 		public readonly int Adjacent = 2;
 
+		[Desc("Type of influences, defined by the `GivesBuildableArea` traits of the nearby actors, this can be build on.")]
+		public readonly HashSet<string> AreaTypes = new HashSet<string>() { "building" };
```

Comment:
This is something I've always been against - magic default values for trait info's properties.

Reply:
Well, @pchote said me to give it a default.

Reply:
I agree that it's not worth supporting an empty set. I don't agree with the magic default value though.
Leaving it empty and adding a `[FieldLoader.Require]` attribute should be enough. That will default the behaviour to "no buildings for you until you fix your missing YAML", which would be enforced by the attribute.

Reply:
I'm unsure about forcing it on Building:. What if it is unbuildable? Why would it need an area type?

Reply:
A fallback value makes a lot more sense here than FieldLoader.Requiring every footprintable "building" (civilian structures, terrain doodads, trees, rocks, lamp posts) for GivesBuildableArea. When RA2 onwards the number of these assets are more than the actually buildable ones.